### ISSUE TITLE
content.cookies.store=false stopped working with 2.0

### ISSUE THREAD

**Issue by jnphilipp** [qutebrowser/qutebrowser#6120]
**Version info**:
qutebrowser v2.0.1
Git commit: 
Backend: QtWebEngine (Chromium 83.0.4103.122)
Qt: 5.15.2

**Does the bug happen if you start with `--temp-basedir`?**:
Needs a session.

**Description**
Cookies are stored with `content.cookies.store` set to `false`.

**How to reproduce**
Set `content.cookies.store` to false, login some, e.g. GitHub. Restart. Still logged in.

**Comment by The-Compiler** [comment]
Not sure if this really was caused by v2.0.x or perhaps by a change in Qt. I'll investigate ASAP.

**Comment by The-Compiler** [comment]
I can confirm it works for me with v1.14.1 but breaks with v2.0.0... Will bisect later today to find out what the culprit is.

**Comment by The-Compiler** [comment]
So I can reproduce this with http://www.whatarecookies.com/cookietest.asp - but for some reason not with the test I wrote (whether I use `max_age` or `expires`):


```diff
diff --git c/tests/end2end/fixtures/webserver.py i/tests/end2end/fixtures/webserver.py
index 4677415f1..81a864c8e 100644
--- c/tests/end2end/fixtures/webserver.py
+++ i/tests/end2end/fixtures/webserver.py
@@ -75,6 +75,7 @@ def _check_status(self):
             '/absolute-redirect': [HTTPStatus.FOUND],
 
             '/cookies/set': [HTTPStatus.FOUND],
+            '/cookies/set-custom': [HTTPStatus.FOUND],
 
             '/500-inline': [HTTPStatus.INTERNAL_SERVER_ERROR],
             '/500': [HTTPStatus.INTERNAL_SERVER_ERROR],
diff --git c/tests/end2end/fixtures/webserver_sub.py i/tests/end2end/fixtures/webserver_sub.py
index b7999148f..265fb7085 100644
--- c/tests/end2end/fixtures/webserver_sub.py
+++ i/tests/end2end/fixtures/webserver_sub.py
@@ -32,6 +32,7 @@
 import threading
 import mimetypes
 import pathlib
+import datetime
 from http import HTTPStatus
 
 import cheroot.wsgi
@@ -199,6 +200,18 @@ def set_cookies():
     return r
 
 
+@app.route('/cookies/set-custom')
+def set_custom_cookie():
+    """Set a cookie with a custom max_age/expires."""
+    r = app.make_response(flask.redirect(flask.url_for('view_cookies')))
+    max_age = flask.request.args.get('max_age')
+    expires = flask.request.args.get('expires')
+    r.set_cookie(key='cookie', value='value',
+                 max_age=int(max_age) if max_age else None,
+                 expires=float(expires) if expires else None)
+    return r
+
+
 @app.route('/basic-auth/<user>/<passwd>')
 def basic_auth(user='user', passwd='passwd'):
     """Prompt the user for authorization using HTTP Basic Auth."""
diff --git c/tests/end2end/test_invocations.py i/tests/end2end/test_invocations.py
index b509e355b..6e3e4ecfb 100644
--- c/tests/end2end/test_invocations.py
+++ i/tests/end2end/test_invocations.py
@@ -490,3 +490,36 @@ def test_service_worker_workaround(
         quteproc_new.ensure_not_logged(message='Removing service workers at *')
     else:
         assert not service_worker_dir.exists()
+
+
+@pytest.mark.parametrize('store', [True, False])
+def test_cookies_store(quteproc_new, request, short_tmpdir, store):
+    # Start test process
+    args = _base_args(request.config) + ['--basedir', str(short_tmpdir)]
+    quteproc_new.start(args)
+    quteproc_new.set_setting('content.cookies.store', str(store))
+
+    import datetime
+    expires = datetime.datetime.now() + datetime.timedelta(minutes=5)
+    quteproc_new.open_path(f'cookies/set-custom?expires={expires.timestamp()}', wait=False)
+    quteproc_new.wait_for_load_finished('cookies')
+
+    # Ensure it's set
+    content = quteproc_new.get_content()
+    data = json.loads(content)
+    assert data == {'cookies': {'cookie': 'value'}}
+
+    # Restart
+    quteproc_new.send_cmd(':quit')
+    quteproc_new.wait_for_quit()
+    quteproc_new.start(args)
+
+    # Get cookies again
+    quteproc_new.open_path('cookies')
+
+    content = quteproc_new.get_content()
+    data = json.loads(content)
+
+    # Make sure it was stored / discarded
+    expected_cookies = {'cookie': 'value'} if store else {}
+    assert data == {'cookies': expected_cookies}

```

**Comment by The-Compiler** [comment]
I was able to bisect this.

- works: 77d5a08b9 Skip cookie filtering test with Qt 5.15.0
- can't test: 2c8127102 webenginesettings: Deduplicate profile init code
- broken: 5782123f6 webenginesettings: Don't special-case cookie policy

**Comment by The-Compiler** [comment]
Ah, turns out my test was working because I was setting the option after qutebrowser was already started - I can reproduce it with the test as well when using `-s` to set it (which is equivalent to setting it in a `config.py` or `autoconfig.yml`).

This is fixed now, a v2.0.x release should follow in a couple of days.
